mm2src/coins/lightning/ln_filesystem_persister.rs

```diff
@@ -353,46 +361,75 @@ impl LightningStorage for LightningFilesystemPersister {
         .await
     }
 
-    async fn get_network_graph(&self, network: Network) -> Result<NetworkGraph, Self::Error> {
+    async fn get_network_graph(&self, network: Network, logger: Arc<LogState>) -> Result<NetworkGraph, Self::Error> {
         let path = self.network_graph_path();
         if !path.exists() {
-            return Ok(NetworkGraph::new(genesis_block(network).header.block_hash()));
+            return Ok(NetworkGraph::new(genesis_block(network).header.block_hash(), logger));
         }
         async_blocking(move || {
             let file = fs::File::open(path)?;
             common::log::info!("Reading the saved lightning network graph from file, this can take some time!");
-            NetworkGraph::read(&mut BufReader::new(file))
+            NetworkGraph::read(&mut BufReader::new(file), logger)
                 .map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e.to_string()))
         })
         .await
     }
 
-    async fn get_scorer(&self, network_graph: Arc<NetworkGraph>) -> Result<Scorer, Self::Error> {
+    async fn get_scorer(&self, network_graph: Arc<NetworkGraph>, logger: Arc<LogState>) -> Result<Scorer, Self::Error> {
         let path = self.scorer_path();
         if !path.exists() {
-            return Ok(Scorer::new(ProbabilisticScoringParameters::default(), network_graph));
+            return Ok(Mutex::new(ProbabilisticScorer::new(
+                ProbabilisticScoringParameters::default(),
+                network_graph,
+                logger,
+            )));
         }
         async_blocking(move || {
             let file = fs::File::open(path)?;
-            Scorer::read(
+            let scorer = ProbabilisticScorer::read(
                 &mut BufReader::new(file),
-                (ProbabilisticScoringParameters::default(), network_graph),
+                (ProbabilisticScoringParameters::default(), network_graph, logger),
             )
-            .map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e.to_string()))
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
+            Ok(Mutex::new(scorer))
         })
         .await
     }
 
-    async fn save_scorer(&self, scorer: Arc<Mutex<Scorer>>) -> Result<(), Self::Error> {
-        let path = self.scorer_path();
+    async fn get_trusted_nodes(&self) -> Result<HashSet<PublicKey>, Self::Error> {
+        let path = self.trusted_nodes_path();
+        if !path.exists() {
+            return Ok(HashSet::new());
+        }
         async_blocking(move || {
-            let scorer = scorer.lock().unwrap();
+            let file = fs::File::open(path)?;
```

Comment:
How about using `mm2_io::fs::read_json`?
https://github.com/KomodoPlatform/atomicDEX-API/blob/23a8ca7f69836bf8d9005258579d6a698b933083/mm2src/mm2_io/src/fs.rs#L175-L185

Please also check where you can also use some functions from the `mm2_io::fs` module

Reply:
Done